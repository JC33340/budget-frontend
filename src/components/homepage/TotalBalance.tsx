import ItemWrapper from '../ItemWrapper';
import ItemHeading from '../ItemHeading';

type OverallBalanceType = {
    balance: number | undefined;
};

const TotalBalance = ({ balance }: OverallBalanceType) => {
    return (
        <ItemWrapper>
            <div
                className="flex flex-col gap-y-4 h-full"
                data-testid="totalBalance-testid"
            >
                <ItemHeading text="Overall balance" />
                <div className="flex h-full items-center justify-center">
                    <p
                        className={`${balance && balance < 0 ? 'text-red' : 'text-green'}  text-6xl`}
                    >
                        Â£ {balance}
                    </p>
                </div>
            </div>
        </ItemWrapper>
    );
};

export default TotalBalance;
