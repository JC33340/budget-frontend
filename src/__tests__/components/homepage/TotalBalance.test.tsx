import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TotalBalance from '../../../components/homepage/TotalBalance';

test('snapshot test', () => {
    render(<TotalBalance balance={100} />);
    const item = screen.getByTestId('totalBalance-testid');
    expect(item).toMatchSnapshot();
});

test('testing content', () => {
    render(<TotalBalance balance={100} />);
    const item = screen.getByTestId('totalBalance-testid');
    expect(item).toHaveTextContent('100');
});
