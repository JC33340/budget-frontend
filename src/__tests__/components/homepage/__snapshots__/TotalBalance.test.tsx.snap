// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="flex flex-col gap-y-4 h-full"
  data-testid="totalBalance-testid"
>
  <p
    class="text-3xl text-blue font-medium text-center"
    data-testid="itemHeading-testid"
  >
    Overall balance
  </p>
  <div
    class="flex h-full items-center justify-center"
  >
    <p
      class="text-green  text-6xl"
    >
      Â£ 
      100
    </p>
  </div>
</div>
`;
